src/Symfony/Component/Console/Command/LockableTrait.php

```diff
@@ -68,4 +78,14 @@ private function release()
             $this->lock = null;
         }
     }
+
+    /**
+     * Sets the internal store for the command to be used.
+     *
+     * @param StoreInterface $store
+     */
+    public function setStore(StoreInterface $store)
```

Comment:
what about passing a non-default store as optional 3rd arg in `lock()`? and avoid this state issue.

Reply:
meaning the user injects his own store if needed

Reply:
If that's the case then this change wouldn't be needed because you can implement the Symfony\Component\Lock\Factory service in a command by dependency injection to create locks.

Reply:
Right ... the user can also set `private $store;` already.Then, im not sure what we're solving here :thinking: 

We should favor pure DI / service usage IMHO.

Reply:
I'm trying to solve a configurable store that can be used in commands, this can either be done by:

- The current solution (setStore in `LockableTrait`);
- Implementing a default store in `Symfony\Component\Console\Command\Command`;
- Writing a `LockableCommand` class which basically does what LockableTrait does now, so this could mean bye bye for LockableTrait - would also mean more abstraction;

Why?

Because currently `LockableTrait` is limited to `SemaphoreStore` and `FlockStore`. Also you can use dependency injection but that would mean that you have to overwrite the constructor. Last, it's not configurable as in available in a command by default.

Reply:
> you can use dependency injection but that would mean that you have to overwrite the constructor.

Is that a real problem? IMHO it's the explicit way.. compared to calling an arbitrary `setStore` method. At least i agree with https://github.com/symfony/symfony/pull/31914/files#r294103123; if we do this we should add a marker interface for it.

Reply:
I agree that we should be using a marker interface. I've updated the PR.